pallets/msa/src/lib.rs

```diff
@@ -911,13 +911,20 @@ impl<T: Config> Pallet<T> {
 			// Increment the key counter
 			<PublicKeyCountForMsaId<T>>::try_mutate(msa_id, |key_count| {
 				// key_count:u8 should default to 0 if it does not exist
-				let incremented_key_count: u8 = *key_count + 1;
-				ensure!(
-					incremented_key_count <= T::MaxPublicKeysPerMsa::get(),
-					Error::<T>::KeyLimitExceeded
-				);
-
-				*key_count = incremented_key_count;
+				let checked_key_count = key_count.checked_add(1);
```

Comment:
use sp_runtime::ArithmeticError;

maybe:

```
key_count.checked_add(1).ok_or(ArithmeticError::Overflow)?
```

Reply:
Since the error gets propagated to the end user of the chain and the key limit is being exceeded in the overflow case, I'd like to preserve the error message to `Error::<T>::KeyLimitExceeded`.  

How about:
`key_count.checked_add(1).ok_or(ArithmeticError::Overflow)?`